tools/fast_pusher_.py

```diff
@@ -98,28 +102,33 @@ def main():
   if not args.config and not args.tarball:
     raise Exception('Either --config or --tarball must be specified.')
 
+  if len(args.digest or []) != len(args.layer or []):
+    raise Exception('--digest and --layer must have matching lengths.')
+
   # If config is specified, use that.  Otherwise, fallback on reading
   # the config from the tarball.
   config = args.config
   if args.config:
+    logging.info('Reading config from %r', args.config)
     with open(args.config, 'r') as reader:
       config = reader.read()
   elif args.tarball:
+    logging.info('Reading config from tarball %r', args.tarball)
     with v2_2_image.FromTarball(args.tarball) as base:
       config = base.config_file()
 
-  if len(args.digest or []) != len(args.layer or []):
-    raise Exception('--digest and --layer must have matching lengths.')
-
-  transport = transport_pool.Http(httplib2.Http, size=_THREADS)
-
   # Resolve the appropriate credential to use based on the standard Docker
   # client logic.
+  logging.info('Loading Docker credentials for repository %r', str(name))
```

Comment:
This seems like a good use for a "tracing" module, should such a thing exist. It might also be able to annotate trace spans with short descriptions that could be exported to external collection systems.